src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java

```diff
@@ -82,7 +85,7 @@ public boolean isActive() {
     }
 
     private void notifyAboutChange(Path path) {
-        listeners.get(path).forEach(FileUpdateListener::fileUpdated);
+        Platform.runLater(() -> listeners.get(path).forEach(FileUpdateListener::fileUpdated));
```

Comment:
This is not a good idea in my eyes. This means it will always notify on the UI thread about a file update. But this is sometomes not intended. I would try to avoid a dependency to the javafx in the DefaultFileUpdateMonitor at all.

Reply:
Hi, thanks for the review, and yea, I felt like the changes in DefaultFileUpdateMonitor were treading on dangerous territory.

The unhandled exception I am getting and which I tried to solve with this is "java.lang.IllegalStateException: Not on FX application thread; currentThread = FileUpdateMonitor".

Would an alternative and safer solution to this be to move the Platform.runLater() to the LatexCitationsTabViewModel.java instead so that it is closer to the problem space?

So the listeners I am adding are:
fileUpdateMonitor.addListenerForFile(selectedDirectory, this::refreshLatexDirectory);

And within the refreshLatexDirectory function, I use the Platform.runlater(); to make sure that this is executed safely on the JavaFX thread?

Reply:
Without fully checking, the code of "FileUpdateListener.fileUpdated" should have the call to Platform.Runlater, because the JavaFX thread thing is a thing happening in the update listener, not in the file monitor. Other file listeners don't need JavaFX updates.